### PR TITLE
update spark connect page based on pr feedback

### PR DESCRIPTION
**Pull request by MrPowers**
Addresses the comments from this PR: https://github.com/apache/spark-website/pull/511

Also rewords some of the language.

### CODE REVIEW

**Review by srowen — APPROVED** [review]

File: site/spark-connect/index.html [hunk 1]
```diff
@@ -182,7 +182,7 @@ <h2 id="high-level-spark-connect-architecture">High-level Spark Connect architec
 GlobalLimit 5
 +- LocalLimit 5
    +- SubqueryAlias spark_catalog.default.some_table
-      +- Relation spark_catalog.default.some_table[character#15,franchise#16] parquet
+      +- UnresolvedRelation spark_catalog.default.some_table
```
Comment (dongjoon-hyun) [inline site/spark-connect/index.html:185]: Nice fix.

File: site/spark-connect/index.html [hunk 2]
```diff
@@ -221,11 +221,11 @@ <h2 id="benefits-of-spark-connect">Benefits of Spark Connect</h2>
 
 <p><strong>Spark Connect workloads are easier to maintain</strong></p>
 
-<p>With the Spark JVM architecture, the client and Spark Driver must run identical software versions.  They need the same Java, Scala, and other dependency versions.  Suppose you develop a Spark project on your local machine, package it as a JAR file, and deploy it in the cloud to run on a production dataset.  You need to build the JAR file on your local machine with the same dependencies used in the cloud.  If you compile the JAR file with Scala 2.13, you must also provision the cluster with a Spark JAR compiled with Scala 2.13.</p>
+<p>With the Spark Classic architecture, the client and Spark Driver must run identical software versions.  They need the same Java, Scala, and other dependency versions.  Suppose you develop a Spark project on your local machine, package it as a JAR file, and deploy it in the cloud to run on a production dataset.  You need to build the JAR file on your local machine with the same dependencies used in the cloud.  If you compile the JAR file with Scala 2.13, you must also provision the cluster with a Spark JAR compiled with Scala 2.13.</p>
```
Comment (dongjoon-hyun) [inline site/spark-connect/index.html]: Let's not introduce a new concept, `Spark Classic`. It's not a widely-accepted community concept and this looks ambiguous to me.

File: site/spark-connect/index.html [hunk 3]
```diff
@@ -234,7 +234,7 @@ <h2 id="benefits-of-spark-connect">Benefits of Spark Connect</h2>
 <ul>
   <li><a href="https://github.com/apache/spark/tree/master/python/pyspark/sql/connect">Spark Connect Python</a></li>
   <li><a href="https://github.com/apache/spark-connect-go">Spark Connect Go</a></li>
-  <li><a href="https://github.com/sjrusso8/spark-connect-rs">Spark Connect Rust</a></li>
+  <li><a href="https://github.com/sjrusso8/spark-connect-rs">Spark Connect Rust</a> (third-party project)</li>
```
Comment (dongjoon-hyun) [inline site/spark-connect/index.html:237]: Thanks.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thank you for the following.

In general, it looks good to me too except one new concept, `Spark Classic`. I'd prefer to avoid new jargons here as much as possible.

File: site/spark-connect/index.html (hunk 2 above)
Comment (MrPowers) [reply site/spark-connect/index.html]: OK, I removed the reference to Spark Classic.

**Comment by MrPowers** [comment]
Thanks for reviewing @dongjoon-hyun.  I removed the references to Spark Classic 😄

**Review by dongjoon-hyun — APPROVED** [review]
Thank you!

**Pull request merged**